src/common/reactNativeProjectHelper.ts

```diff
@@ -10,53 +10,111 @@ import {Package} from "./node/package";
-                }
-                return rootProjectPackageJson.devDependencies()
-                    .then(devDependencies => {
-                        if (devDependencies["react-native"]) {
-                            return ReactNativeProjectHelper.processVersion(devDependencies["react-native"]);
+    public static getReactNativeVersionsFromProjectPackage(cwd: string, isRNWVersion: boolean = false): Q.Promise<RNPackageVersions> {
+        let parsedPackageNames: ParsedPackageName[] = [
+            {
+                packageName: "react-native",
+                useSemverCoerce: true,
+            },
+        ];
+
+        if (isRNWVersion) {
+            parsedPackageNames.push({
+                packageName: "react-native-windows",
+                useSemverCoerce: false,
+            });
+        }
+
+        let versionPromises: Q.Promise<string>[] = [];
+
+        parsedPackageNames.forEach(parsedPackageName => {
+            versionPromises.push((() => {
+                const rootProjectPackageJson = new Package(cwd);
+                return rootProjectPackageJson.dependencies()
+                    .then(dependencies => {
+                        if (dependencies[parsedPackageName.packageName]) {
+                            return ReactNativeProjectHelper.processVersion(dependencies[parsedPackageName.packageName], parsedPackageName.useSemverCoerce);
                         }
+                        return rootProjectPackageJson.devDependencies()
+                            .then(devDependencies => {
+                                if (devDependencies[parsedPackageName.packageName]) {
+                                    return ReactNativeProjectHelper.processVersion(devDependencies[parsedPackageName.packageName], parsedPackageName.useSemverCoerce);
+                                }
+                                return "";
+                            });
+                    })
+                    .catch(err => {
                         return "";
                     });
-            })
-            .catch(err => {
-                return "";
-            });
+                })()
+            );
+        });
+
+        return Q.all(versionPromises).then(packageVersions => ({
+            reactNativeVersion: packageVersions[0],
+            reactNativeWindowsVersion: packageVersions[1] || "",
+        }));
     }
 
-    public static processVersion(version: string): string {
+    public static processVersion(version: string, useSemverCoerce: boolean = true): string {
         try {
             return new URL(version) && "SemverInvalid: URL";
         } catch (err) {
-            const versionObj = semver.coerce(version);
+            let versionObj;
+            if (useSemverCoerce) {
+                versionObj = semver.coerce(version);
+            } else {
+                versionObj = semver.clean(version.replace(/[\^~]/g, ""), { loose: true });
```

Comment:
Please make sure that we consider all possible critical specsymbols.

Reply:
I added symbols `<`, `>` to the regex because `semver.clean` function doesn't operate them correctly (If there is any of these symbols in a version string, the `clean` function returns `null`).

Reply:
What about asterisk symbol?